test/escaping.js

```diff
@@ -1,30 +1,19 @@
 var test = require('tap').test,
     ecstatic = require('../'),
     http = require('http'),
-    request = require('request');
-
-var server;
+    request = require('request'),
+    eol = require('eol');
 
 test('escaping special characters', function (t) {
-  server = http.createServer(ecstatic(__dirname + '/public'));
+  var server = http.createServer(ecstatic(__dirname + '/public'));
  
   server.listen(0, function () {
     var port = server.address().port;
     request.get('http://localhost:' + port + "/curimit%40gmail.com%20(40%25)", function (err, res, body) {
       t.ifError(err);
       t.equal(res.statusCode, 200);
-      t.equal(body, 'index!!!\n');
-      t.end();
+      t.equal(eol.lf(body), 'index!!!\n');
+      server.close(function() { t.end(); });
```

Comment:
yes, as far as I can tell. I just switch the logic around. Instead of closing the server on t.end, I call t.end() when the server is closed.